kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/CsrServiceImpl.java

```diff
@@ -320,6 +326,29 @@ private CallSummaryRecord aggregateSummaryRecord(CallSummaryRecordDto csr) {
     }
 
 
+    private void validateCallSummaryRecord(Subscription subscription, CallSummaryRecordDto csr) throws InvalidCdrData {
+        if (!subscription.getSubscriptionPack().hasMessageWithWeekId(csr.getWeekId())) {
+            throw new InvalidCdrData(String.format("%s is an invalid weekId for pack %s",
+                    csr.getWeekId(), subscription.getSubscriptionPack().getName()));
+        }
+
+        if (!subscription.getSubscriptionPack().hasMessageWithFilename(csr.getContentFileName())) {
+            throw new InvalidCdrData(String.format("%s is an invalid contentFilename for pack %s",
+                    csr.getContentFileName(), subscription.getSubscriptionPack().getName()));
+        }
+
+        if (circleService.getByName(csr.getCircle()) == null) {
```

Comment:
Don't we tell them what the circle is and they just return it back?  If we don't get a circle do we still want to record the call?